packages/ai-constructs/src/conversation/runtime/bedrock_converse_adapter.ts

```diff
@@ -153,6 +157,169 @@ export class BedrockConverseAdapter {
     return bedrockResponse.output?.message?.content ?? [];
   };
 
+  /**
+   * Asks Bedrock for response using streaming version of Converse API.
+   */
+  async *askBedrockStreaming(): AsyncGenerator<StreamingResponseChunk> {
+    const { modelId, systemPrompt, inferenceConfiguration } =
+      this.event.modelConfiguration;
```

Comment:
This is the core of new functionality.

Inspired by https://docs.aws.amazon.com/bedrock/latest/userguide/tool-use-examples.html (make sure to look at `ConverseStream` section). 

It is using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function* and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator .
to represent stream of asynchronous chunks.